### PR TITLE
Add SaveChat & Chats endpoint

### PR DESCRIPTION
**Pull request by cammellos**
Add endpoints to build store/retrieve chats.
Please do not merge even if approved, as I would like it to go through a round of testing frist.

Reverts commit https://github.com/status-im/status-go/commit/476240fbd107fe193096350f5181c32c0539eff0 as I am unable to login in status-react with it, it might require changes in `status-react`, if those get merged before this is ready to be merged, I will rebase my branch, otherwise I will revert and merge this one.

### CODE REVIEW

File: node/status_node_test.go [hunk 1]
```diff
@@ -48,8 +53,11 @@ func TestStatusNodeStart(t *testing.T) {
 	accountManager, err := n.AccountManager()
 	require.Nil(t, err)
 	require.NotNil(t, accountManager)
+	keyStore, err := n.AccountKeyStore()
+	require.Nil(t, err)
```
Comment (pedropombeiro) [inline node/status_node_test.go:57]:
```suggestion
	require.NoError(t, err)
```

**Review by pedropombeiro — APPROVED** [review]
lgtm, just a minor comment

File: node/status_node_test.go (hunk 1 above)
Comment (cammellos) [reply node/status_node_test.go:57]: thanks, I won't apply this as this is part of the revert commit, I believe it will be fixed once we re-apply

**Comment by cammellos** [comment]
@dshulyak @yenda I have merged the revert, feel free to re-open a PR with the changes, thanks!

**Pull request merged**